Src/LogData.c

```diff
@@ -10,24 +19,52 @@
 #include "Data.h"
 #include "FlightPhase.h"
 
+
+/* Structs -------------------------------------------------------------------*/
+typedef struct{
+    int32_t accelX;
+    int32_t accelY;
+    int32_t accelZ;
+    int32_t gyroX;
+    int32_t gyroY;
+    int32_t gyroZ;
+    int32_t magnetoX;
+    int32_t magnetoY;
+    int32_t magnetoZ;
+    int32_t barometerPressure;
+    int32_t barometerTemperature;
+    int32_t combustionChamberPressure;
+    int32_t oxidizerTankPressure;
+    int32_t gps_time;
+    int32_t latitude_degrees;
+    int32_t latitude_minutes;
+    int32_t longitude_degrees;
+    int32_t longitude_minutes;
+    int32_t antennaAltitude;
+    int32_t geoidAltitude;
+    int32_t altitude;
+    int32_t currentFlightPhase;
+    int32_t tick;
+} logEntry; // logEntry holds data from AllData that is to be logged
+
+
+/* Constants -----------------------------------------------------------------*/
 static int SLOW_LOG_DATA_PERIOD = 700;
 static int FAST_LOG_DATA_PERIOD = 200;
 static uint8_t softwareVersion = 104;
 static uint8_t deviceAddress = 0x50;
-static uint8_t currMemAddress = 0x07;
-
-static FATFS fatfs;
-static FIL file;
-
-char fileName[32];
+static uint8_t currMemAddress = 0x07; //TODO: update this as needed
+static uint32_t timeout = 10; //Time required to wait before ending read/write (unit =?)
+static logEntry theLogEntry; //This is the global log entry that is updated
+static uint8_t logEntrySize = 84;
```

Comment:
Can we not use `sizeof(logEntry)` instead of having a constant? The `sizeof(...)` function gets evaluated at compile time, so no speed penalty there. Also, unless some of the types in the `logData` struct need to be changed, then this value is wrong (i.e. a `logData` struct is currently 23*uint32_t, which is 96 bytes).